h/cli/commands/authclient.py

```diff
@@ -13,14 +14,18 @@ def authclient():
 @authclient.command()
 @click.option('--name', prompt=True, help="The name of the client")
 @click.option('--authority', prompt=True, help="The authority (domain name) of the resources managed by the client")
+@click.option('--type', 'type_', type=click.Choice(['public', 'confidential']), prompt=True, help="The OAuth client type (public, or confidential)")
 @click.pass_context
-def add(ctx, name, authority):
+def add(ctx, name, authority, type_):
     """
     Create a new OAuth client.
     """
+
     request = ctx.obj['bootstrap']()
 
     authclient = models.AuthClient(name=name, authority=authority)
+    if type_ == 'confidential':
+        authclient.secret = token_urlsafe()
```

Comment:
This would make more sense in a service, but since we already manually create the auth client in here and it's the only place where we currently create auth clients, I've opted for not pulling this out into a new service.